src/query/exist_query.rs

```diff
@@ -0,0 +1,287 @@
+use core::fmt::Debug;
+
+use columnar::{ColumnType, DynamicColumn};
+
+use super::range_query::VecCursor;
+use super::ConstScorer;
+use crate::core::SegmentReader;
+use crate::docset::{DocSet, TERMINATED};
+use crate::query::explanation::does_not_match;
+use crate::query::{EnableScoring, Explanation, Query, Scorer, Weight};
+use crate::schema::FieldType;
+use crate::{DocId, Score, TantivyError};
+
+/// Query that matches all of the documents.
+///
+/// All of the document get the score 1.0.
+#[derive(Clone, Debug)]
+pub struct ExistsQuery {
+    field: String,
+}
+
+impl ExistsQuery {
+    /// Creates a new `RangeQuery` from bounded start and end terms.
+    ///
+    /// If the value type is not correct, something may go terribly wrong when
+    /// the `Weight` object is created.
+    pub fn new_exists_query(field: String) -> ExistsQuery {
+        ExistsQuery { field }
+    }
+}
+
+impl Query for ExistsQuery {
+    fn weight(&self, enable_scoring: EnableScoring<'_>) -> crate::Result<Box<dyn Weight>> {
```

Comment:
Nitpick: I suspect you can drop the lifetime here. (no?)

```suggestion
    fn weight(&self, enable_scoring: EnableScoring) -> crate::Result<Box<dyn Weight>> {
```